File: backend/schemas/job.py
from pydantic import BaseModel, HttpUrl,Field
from typing import List, Optional

class Job(BaseModel):
    """
    Represents a single job posting.
    """
    title: str
    company: str
    location: str
    employment_type: Optional[str] = None
    experience: Optional[str] = None
    salary: Optional[str] = None
    skills: List[str] = Field(default_factory=list)
    description: str
    apply_url: HttpUrl
    source: str

File: frontend/src/api/job.ts
import api from "./axios";

export interface Job {
  title: string;
  company: string;
  location: string;
  employment_type?: string | null;
  experience?: string | null;
  salary?: string | null;
  skills: string[];
  description: string;
  apply_url?: string | null;
  source: string;
  match_score?: number | null;
}

export interface RecommendedJobsResponse {
  success: boolean;
  data: Job[];
}

export const getRecommendedJobs = async (): Promise<Job[]> => {
  const response = await api.get<RecommendedJobsResponse>(
    "/api/jobs/recommended"
  );

  return response.data.data;
};